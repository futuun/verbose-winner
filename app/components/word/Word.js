import React, { PropTypes } from 'react'
import styles from './styles.sass'
import Letter from '../letter'

function Word(props) {
  const {
    exposedLetters,
    secretWord,
  } = props
  const leftpadSecretWord = (`-----------${secretWord}`).slice(-11)

  return (
    <div className={styles.word}>
      {leftpadSecretWord.split('').map((letter, i) => (
        <Letter
          key={i}
          value={letter}
          exposed={exposedLetters.includes(letter)}
          inactive={letter === '-'}
        />
      ))}
    </div>
  )
}

Word.defaultProps = {
  secretWord: '',
  exposedLetters: [],
}

Word.propTypes = {
  exposedLetters: PropTypes.arrayOf(PropTypes.string).isRequired,
  secretWord: PropTypes.string.isRequired,
}

export default Word
