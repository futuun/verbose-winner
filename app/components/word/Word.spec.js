import React from 'react'
import renderer from 'react-test-renderer'
import Word from './Word'

test('Word: no exposed letters', () => {
  const component = renderer.create(
    <Word
      secretWord="kopytko"
      exposedLetters={[]}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Word: 1 letter exposed, 2 occurrences', () => {
  const component = renderer.create(
    <Word
      secretWord="kopytko"
      exposedLetters={['k']}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Word: multiple exposed letters', () => {
  const component = renderer.create(
    <Word
      secretWord="kopytko"
      exposedLetters={['k', 'y', 'o']}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
