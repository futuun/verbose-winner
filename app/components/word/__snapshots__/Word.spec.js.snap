exports[`test Word: 1 letter exposed, 2 occurrences 1`] = `
<div
  className="word">
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter">
    k
  </div>
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter">
    k
  </div>
  <div
    className="letter" />
</div>
`;

exports[`test Word: multiple exposed letters 1`] = `
<div
  className="word">
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter">
    k
  </div>
  <div
    className="letter">
    o
  </div>
  <div
    className="letter" />
  <div
    className="letter">
    y
  </div>
  <div
    className="letter" />
  <div
    className="letter">
    k
  </div>
  <div
    className="letter">
    o
  </div>
</div>
`;

exports[`test Word: no exposed letters 1`] = `
<div
  className="word">
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter inactive" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
  <div
    className="letter" />
</div>
`;
